pkg/expr/sql/db_test.go

```diff
@@ -205,6 +206,80 @@ func TestErrorsFromGoMySQLServerAreFlagged(t *testing.T) {
 	require.Contains(t, err.Error(), "error in go-mysql-server")
 }
 
+func TestFrameToSQLAndBack_JSONRoundtrip(t *testing.T) {
+	expectedFrame := &data.Frame{
+		RefID: "json_test",
+		Name:  "json_test",
+		Fields: []*data.Field{
+			data.NewField("id", nil, []*int64{p(int64(1)), p(int64(2))}),
+			data.NewField("payload", nil, []*json.RawMessage{
+				p(json.RawMessage(`{"foo":1}`)),
+				p(json.RawMessage(`{"bar":"baz"}`)),
+			}),
+		},
+	}
+
+	sqlDB := &DB{}
+
+	query := `SELECT * FROM json_test`
+
+	resultFrame, err := sqlDB.QueryFrames(context.Background(), "json_test", query, data.Frames{expectedFrame})
+	require.NoError(t, err)
+
+	// Ensure consistent names for comparison
+	resultFrame.Name = expectedFrame.Name
+	resultFrame.RefID = expectedFrame.RefID
```

Comment:
Not the hugest fan of this part, I'll be honest. I'd rather simply test for the things that we do expect to match, rather than mutating the result from the database before we do a comparison.

~~We're passing a `data.FrameTestCompareOptions()...` here, right? Could we tell cmp.Diff to ignore these fields with that, maybe 🤔 ?~~

I would personally omit `Name` and `RefID` from expectedFrame, or set them to what we're actually expecting to get back, when we create that object. And maybe the object at the top should be named `InputFrame`

Same comment applies to the next test in this file too.